pandas/tests/test_categorical.py

```diff
@@ -173,6 +173,12 @@ def test_constructor_unsortable(self):
         pytest.raises(
             TypeError, lambda: Categorical(arr, ordered=True))
 
+    def test_constructor_dataframe_error(self):
+        # GH#17112
+        df = pd.DataFrame(np.random.randn(3, 2), columns=['A', 'B'])
+        with pytest.raises(NotImplementedError):
```

Comment:
This is where checking the error message is helpful.  AFAICT, your tests don't cover both places where a `NotImplementedError` can be raised.

Reply:
agree here

Reply:
Good idea, will do.